### ISSUE TITLE
menuPerm list problem

### ISSUE THREAD

**Issue by LeonaMorro** [HowardBaxton/nPose#22]
Under some conditions a NC will not be parsed. That is also reproducable in Gecko Release
# Setup

NC name: `BTN:Test1:Test{owner}`
content: `LINKMSG|5000|%CARDNAME%`

NC name `BTN:Test2:Test`
content: `LINKMSG|5000|%CARDNAME%`
# Problem

pressing the `Test2:Test` button inside the menu will not trigger `DOACTION`
# Details

In the MenuPerm list only the buttonName and permission ist stored. For our setup something like this:
`Test, owner, Test,`
If I press the button in path Test2, we do a `llListFindList(menuPerm, ["Test"])` which of course only finds the first entry.
So we use `BTN:Test2:Test{owner}` for the inventory lookup. Thats of course wrong.

**Comment by LeonaMorro** [comment]
I'm quite sure that this bug has the potential for many subsequent weird problems. Let us imagine we add a third button to the above setup:
`BTN:Test3:owner`
that would lead to 
menuPerm list: `Test, owner, Test, , owner,`
and an inventory search for:
`BTN:Test3:owner{Test}`

I guess instead of doing a `llListFindList` we have to loop through the menuPerm list.

**Comment by LeonaMorro** [comment]
Even the bugfix I made may fail under some conditions. For example:
`SET:example{owner}`
`SET:owner{something}`
will produce this menuPerm list:
`example, owner, owner, something`
If we press the button `owner` we do a lookup for the NC: `SET:owner{owner}` which if of course wrong.
(I currently can't test the above statement inWorld, because there are some issues in SL that didn't allow me to rezz anything)

I see two possible solutions:
1) using the whole path in menuPerm list incl. the `Main:` (currently we strip the first part)
2) separate the menuPerm list into two lists for example: menuPermPath + menuPermPerms

The variant 2) seems to be a much cleaner approach and also uses less memory. I'm going to make an example for it.

**Comment by HowardBaxton** [comment]
Is there a possibility to grab a sub list to do the llListFindList?  I am referring to http://wiki.secondlife.com/wiki/LlList2ListStrided

**Comment by LeonaMorro** [comment]
Yes of course, we can do something like that:

```
index=llListFindList(llList2ListStrided(menuPerm, 0, -1, 2), PATH);
index=index*2
permissions=llList2String(menuPerm, index+1)
```

but in fact `llList2ListStrided(menuPerm, 0, -1, 2)` will generate a new list which has every second element of menuPerm as list elements. I don't know if the above example counts against the memory limit, but I also see no advantage in comparison with the menuPermPath+menuPermPerms variant.

**Comment by LeonaMorro** [comment]
Fixed in 2cc76a7b962b855e9e593140a5a5128910168b44
